### PR TITLE
Fix printing of stacktrace with unknown command

### LINKED ISSUE #4256: Suggestion of command prints stacktrace
In the latest master, running for example `java -jar build/libs/gatk.jar FixVcfHead` returns:

```
USAGE:  <program name> [-h]

Available Programs:
--------------------------------------------------------------------------------------
Base Calling:                                    Tools that process sequencing machine data, e.g. Illumina base calls, and detect sequencing level attributes, e.g. adapters
    CheckIlluminaDirectory (Picard)              Asserts the validity for specified Illumina basecalling data.
    CollectIlluminaBasecallingMetrics (Picard)   Collects Illumina Basecalling metrics for a sequencing run.

...skipped for brevity...

    VcfFormatConverter (Picard)                  Converts VCF to BCF or BCF to VCF.
    VcfToIntervalList (Picard)                   Converts a VCF or BCF file to a Picard Interval List

--------------------------------------------------------------------------------------

Exception in thread "main" org.broadinstitute.hellbender.exceptions.UserException: 'FixVcfHead' is not a valid command.
Did you mean this?
        FixVcfHeader
	at org.broadinstitute.hellbender.Main.extractCommandLineProgram(Main.java:341)
	at org.broadinstitute.hellbender.Main.setupConfigAndExtractProgram(Main.java:172)
	at org.broadinstitute.hellbender.Main.mainEntry(Main.java:192)
	at org.broadinstitute.hellbender.Main.main(Main.java:275)
```

I expect something without the stack trace and the scary "Exception" message. For example:

```
USAGE:  <program name> [-h]

Available Programs:
--------------------------------------------------------------------------------------
Base Calling:                                    Tools that process sequencing machine data, e.g. Illumina base calls, and detect sequencing level attributes, e.g. adapters
    CheckIlluminaDirectory (Picard)              Asserts the validity for specified Illumina basecalling data.
    CollectIlluminaBasecallingMetrics (Picard)   Collects Illumina Basecalling metrics for a sequencing run.

...skipped for brevity...

    VcfFormatConverter (Picard)                  Converts VCF to BCF or BCF to VCF.
    VcfToIntervalList (Picard)                   Converts a VCF or BCF file to a Picard Interval List

--------------------------------------------------------------------------------------

Did you mean this?
        FixVcfHeader
```

The same happens with unknown commands. The code that should be changed for that is the following, where the `setupConfigAndExtractProgram` call should be also inside the try block:

https://github.com/broadinstitute/gatk/blob/8ac2f102b303f343c4787ad4e3359335641c5121/src/main/java/org/broadinstitute/hellbender/Main.java#L190-L212

### PR DESCRIPTION
**Pull request by magicDGS**
Closes #4256

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/Main.java [hunk 1]
```diff
@@ -65,7 +66,8 @@
     /**
      * exit value when an unrecoverable {@link UserException} occurs
      */
-    private static final int USER_EXCEPTION_EXIT_VALUE = 2;
+    @VisibleForTesting
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/Main.java]: It's fine to just make this public.

Comment (magicDGS) [reply src/main/java/org/broadinstitute/hellbender/Main.java]: Done.

File: src/test/java/org/broadinstitute/hellbender/MainTest.java [hunk 2]
```diff
@@ -50,4 +51,46 @@ public void testClpOmitFromCommandLine() {
         Assert.assertFalse(usage.contains(clpName));
     }
 
+
+    private static final class ExitNotAllowedExcepion extends SecurityException {
```
[the remark this answers is not in the document]
Comment (magicDGS) [reply src/test/java/org/broadinstitute/hellbender/MainTest.java]: Done.

File: src/test/java/org/broadinstitute/hellbender/MainTest.java [hunk 3]
```diff
@@ -50,4 +51,46 @@ public void testClpOmitFromCommandLine() {
...
+        final int status;
+
+        ExitNotAllowedExcepion(int status) {
+            this.status = status;
+        }
+    }
+
+    private static final class NoAllowExitSecurityManager extends SecurityManager {
```
[the remark this answers is not in the document]
Comment (magicDGS) [reply src/test/java/org/broadinstitute/hellbender/MainTest.java]: Done.

File: src/test/java/org/broadinstitute/hellbender/MainTest.java [hunk 4]
```diff
@@ -50,4 +51,46 @@ public void testClpOmitFromCommandLine() {
...
+
+        @Override
+        public void checkExit(int status) {
+            super.checkExit(status);
+            // always throw
+            throw new ExitNotAllowedExcepion(status);
+        }
+    }
+
+    @Test(singleThreaded = true)
```
[the remark this answers is not in the document]
Comment (magicDGS) [reply src/test/java/org/broadinstitute/hellbender/MainTest.java]: Thanks!

**Comment by magicDGS** [comment]
@lbergelson - back to you!

I agree that the tests are a bit complicated, but the only way of testing this. In addition, the new static classes can be used to test other error codes, and thus it will be useful for testing `mainEntry` in another PR for other error codes.

**Comment by lbergelson** [comment]
@magicDGS Thank you for the update, I noticed one last minor thing I wanted to change so I just made the change myself, I'll merge when the test finish!

**Comment by magicDGS** [comment]
Thanks @lbergelson

**Pull request merged**